### DISCUSSION TITLE
support bigint / int64

### DISCUSSION THREAD

**Discussion by derekperkins** [meilisearch/product#684] [Feedback & Feature Proposal]
We have several integer ids that we'd like to store as integers in Meilisearch, but have to use strings for because of int53 max.

**Comment by macraig** [comment]
Hi @derekperkins , thanks for taking the time to leave your feedback! I'm assuming that your issue is that your integers are larger than int53. If so, how large are they? And if not, can you please expand on your use case? Thank you!

**Reply by macraig** [reply]
@derekperkins quick update here: I checked with the team and we do support int64. Are you running into issues trying to use them?

**Reply by derekperkins** [reply]
> Limitation: Meilisearch can only exactly represent integers between -2⁵³ and 2⁵³.
> https://www.meilisearch.com/docs/learn/advanced/known_limitations#size-of-integer-fields

According to the docs, they aren't supported so we haven't tried. The use case is that all of our ids are int64, with the vast majority > 2^53

**Reply by macraig** [reply]
@derekperkins you're right. I re-checked and the docs are correct, there is in fact an int53 limitation. I'm really sorry about the confusion! Could you tell me a bit more about why it's an issue to use Strings?

**Reply by derekperkins** [reply]
No major issues, storing as strings instead of integers presumably utilizes significantly more storage space and cpu for no particular benefit. We have a lot of reference ids being stored. Not a high priority issue, but worth considering.

**Reply by macraig** [reply]
Thank you for expanding @derekperkins, I'll keep this discussion open for other users to upvote.
